test-helpers/src/connection/kafka/mod.rs

```diff
@@ -162,10 +162,13 @@ impl KafkaConsumer {
             expected_response.key, response.key,
             "Unexpected key for topic {topic}"
         );
-        assert_eq!(
-            expected_response.offset, response.offset,
-            "Unexpected offset for topic {topic}"
-        );
+
+        if expected_response.offset.is_some() {
```

Comment:
The documentation and PartialEq implementation of ExpectedResponse already match this logic, but we missed it here somehow.